api/lib/application/campaigns/campaign-controller.js

```diff
@@ -67,9 +67,25 @@ module.exports = {
   update(request) {
     const { userId } = request.auth.credentials;
     const campaignId = parseInt(request.params.id);
-    const { title, 'custom-landing-page-text': customLandingPageText } = request.payload.data.attributes;
+    const { title, 'custom-landing-page-text': customLandingPageText, 'archived-at': archivedAt } = request.payload.data.attributes;
 
-    return usecases.updateCampaign({ userId, campaignId, title, customLandingPageText })
+    return usecases.updateCampaign({ userId, campaignId, title, customLandingPageText, archivedAt })
+      .then(campaignSerializer.serialize);
+  },
+
+  archiveCampaign(request) {
+    const { userId } = request.auth.credentials;
+    const campaignId = parseInt(request.params.id);
+
+    return usecases.updateCampaign({ userId, campaignId, archivedAt: new Date() })
```

Comment:
normalement PG est capable de faire un "new Date()" tout seul non ? en le spécifiant comme `hasTimestamps` dans infrastructure/data/campaign.js (le modèle Bookshelf de Campaign)

Actuellement il y a `hasTimestamps: ['createdAt', null]`, je suis pas sûre de l'utilité du `null` par contre ?

Reply:
Je ne suis pas sûr de comprendre ce que tu voudrais faire / les avantages que ça aurait ? Il s'agit bien de fixer la date	actuelle sur un update ici, ce n'est pas comme un `createdAt` qui se mettrait une fois automatiquement à la création. Je ne sais pas non plus ce que veut dire tous ces : 
```
hasTimestamps: ['createdAt', null]
```
Mais on en a partout.

Reply:
Dites donc, en désespoir de cause vous pouvez lire la doc : https://bookshelfjs.org/api.html#Model-instance-hasTimestamps 😏

Reply:
`For example, to automatically update only the created_at attribute set this property to ['created_at', null].`
Merci jonathan :)